workers/generic-worker/loopback_video_posix.go

```diff
@@ -0,0 +1,90 @@
+//go:build darwin || linux || freebsd
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/taskcluster/taskcluster/v53/internal/scopes"
+	"github.com/taskcluster/taskcluster/v53/workers/generic-worker/host"
+)
+
+type LoopbackVideoFeature struct {
+}
+
+func (feature *LoopbackVideoFeature) Name() string {
+	return "LoopbackVideo"
+}
+
+func (feature *LoopbackVideoFeature) Initialise() error {
+	return nil
+}
+
+func (feature *LoopbackVideoFeature) PersistState() error {
+	return nil
+}
+
+func (feature *LoopbackVideoFeature) IsEnabled(task *TaskRun) bool {
+	return task.Payload.Features.LoopbackVideo
+}
+
+func (feature *LoopbackVideoFeature) NewTaskFeature(task *TaskRun) TaskFeature {
+	return &LoopbackVideoTask{
+		task: task,
+	}
+}
+
+type LoopbackVideoTask struct {
+	task *TaskRun
+}
+
+func (lvt *LoopbackVideoTask) RequiredScopes() scopes.Required {
+	return scopes.Required{{
+		"generic-worker:loopback-video:" + config.ProvisionerID + "/" + config.WorkerType,
+	}}
+}
+
+func (lvt *LoopbackVideoTask) ReservedArtifacts() []string {
+	return []string{}
+}
+
+func (lvt *LoopbackVideoTask) Start() *CommandExecutionError {
+	if runtime.GOOS == "darwin" {
+		return MalformedPayloadError(errors.New("Loopback video is not supported on darwin"))
```

Comment:
Thanks for adding this.

In general I think in taskcluster we refer to the platform as macOS, so I think "Loopback video is not supported on macOS" would be more consistent with our other messages (although I haven't double checked all of them). e.g. we provide setup instructions for macOS.

This feature also won't work with freebsd I guess (I think v4l2loopback is a linux only thing). If we added support, the implementation would be different for linux/freebsd/macOS so I think I would prefer it if you rename this file to loopback_video_linux.go and create a stub for the other platforms that just return a MalformedPayloadError and have nothing else. Then, if we later add support, it will go in its own file, and if we don't, it will be clear to anyone looking at the code that it is linux only (they won't have to _notice_ that there is an if statement on line 55 - because when they go to open the code, they will see immediately there is a linux version, a freebsd and a macOS version).